src/ServiceManager.php

```diff
@@ -562,7 +560,7 @@ private function getFactory(string $name): callable
      */
     private function createDelegatorFromName(string $name, ?array $options = null)
     {
-        $creationCallback = function () use ($name, $options) {
+        $creationCallback = function () use ($name, $options): object {
```

Comment:
This cannot be `: object` IMO: a delegator could produce anything

Reply:
Ok, rolled back https://github.com/laminas/laminas-servicemanager/pull/157/commits/0c9036e4d78bbbbe9c9c26a7c5fc05e11d77f6f8